pkg/discovery/oci_dbbacked.go

```diff
@@ -244,16 +248,15 @@ func (od *DBBackedOCIDiscovery) downloadInventoryDatabase() error {
 // It returns an empty string if the cache can be used.  Otherwise
 // it returns the name of the digest file that must be created once
 // the new DB image has been downloaded.
-func (od *DBBackedOCIDiscovery) checkImageCache() (string, string) {
+func (od *DBBackedOCIDiscovery) checkImageCache() (string, string, error) {
 	// Get the latest digest of the discovery image.
 	// If the cache already contains the image with this digest
 	// we do not need to verify its signature nor to download it again.
 	_, hashHexValInventoryImage, err := carvelhelpers.GetImageDigest(od.image)
 	if err != nil {
 		// This will happen when the user has configured an invalid image discovery URI
 		log.Warningf("Unable to resolve the plugin discovery image: %v", err)
-		// We force abort execution here to make sure a stale image left in the cache is not used by mistake.
-		log.Fatal(nil, fmt.Sprintf("Fatal: plugins discovery image resolution failed. Please check that the repository image URL %q is correct ", od.image))
+		return "", "", fmt.Errorf("plugins discovery image resolution failed. Please check that the repository image URL %q is correct ", od.image)
```

Comment:
Do we need to restrict the user here? Considering the user added a registry that was accessible when it was added.
Now, it is no longer accessible for some reason. Do we want to stop users from running some of the plugin commands like search?

I feel this validation should only be done as part of updating the discovery source and we can keep the existing behavior of throwing an warning as part of `checkImageCache`.